functions.php

```diff
@@ -28,14 +38,59 @@ function DBCheck($data): string {
 
 function makePlantEntryTile(array $entry): string {
     if (empty($entry)) {
-        return 'There is no data for this entry';
+        $return_string = 'There is no data for this entry';
     } elseif (count($entry) < 4) {
-        return 'There is not enough information given for this type of entry.';
+        $return_string = 'There is not enough information given for this type of entry.';
     } elseif (!empty($entry)) {
-        return "<div class='entry_box'>
+        $return_string = "<div class='entry_box'>
                         <div class='entry science_name'>" . $entry['science_name'] . "</div>
-                        <div class='entry'>" . $entry['name'] . "</div>
-                        <div class='entry'>" . $entry['type'] . "</div>
+                        <div class='entry'>" . ucwords($entry['name'],"/ -") . "</div>
+                        <div class='entry'>" . ucwords(strtolower($entry['type']),"/ -") . "</div>
                     </div>";
     }
+
+    return $return_string;
+}
+
+function listPlantTypes(array $plant_types): string {
+    if (array_key_exists('type',$plant_types[0])) {
+        $type_echo = '<ul>';
+        foreach ($plant_types as $entry) {
+            $type_echo.= '<li>' . ucwords(strtolower($entry['type']),"/ -") . '</li>';
+        }
+        $type_echo.= '</ul>';
+    } else {
+        $type_echo = 'Array key error. Please enter different data.';
+    }
+
+    return $type_echo;
+}
+
+function checkforUniqueAddEntry($db, string $science_name): bool {
+    $nonunique_entry_query = $db->prepare("SELECT `id` FROM `plants` WHERE `science_name` = ?;");
+    $nonunique_entry_query->execute([$science_name]);
+    $nonunique_entry = $nonunique_entry_query->fetch();
+    if ($nonunique_entry !== false) {
+        $nonunique_entry = true;
+    }
+
+    return $nonunique_entry;
 }
+
+function checkForWrongPlantType($db, string $type): array {
```

Comment:
this function does two things. it gets the plant id AND checks if its valid. the first thing is actually the more important. as a plant can only have 1 type i would make this function return an int (the plant id value), then you can make it return 0 (which equates to false) if its an incorrect type, which will make doing an `if` on it easier